unlock-app/src/components/interface/keychain/TransferKeyDrawer.tsx

```diff
@@ -0,0 +1,170 @@
+import { z } from 'zod'
+import { onResolveName } from '~/utils/resolvers'
+import { useWeb3Service } from '~/utils/withWeb3Service'
+import { ethers } from 'ethers'
+import { useWalletService } from '~/utils/withWalletService'
+import { ToastHelper } from '~/components/helpers/toast.helper'
+import { addressMinify } from '~/utils/strings'
+import { useMutation } from '@tanstack/react-query'
+import { useLockData } from '~/hooks/useLockData'
+import { useGetTransferFeeBasisPoints } from '~/hooks/useTransferFee'
+import { useEffect } from 'react'
+
+interface TransferKeyDrawerProps {
+  isOpen: boolean
+  setIsOpen: (open: boolean) => void
+  lockAddress: string
+  network: number
+  tokenId: string
+  owner: string
+  lockName?: string
+}
+
+const TransferKeyForm = z.object({
+  newOwner: z.string({
+    description: 'Address of the new owner.',
+  }),
+})
+
+type TransferKeyFormProps = z.infer<typeof TransferKeyForm>
+
+export const TransferKeyDrawer = ({
+  isOpen,
+  setIsOpen,
+  lockAddress,
+  network,
+  tokenId,
+  owner,
+  lockName,
+}: TransferKeyDrawerProps) => {
+  const {
+    watch,
+    control,
+    setValue,
+    handleSubmit,
+    formState: { errors },
+    reset,
+  } = useForm<TransferKeyFormProps>({
+    defaultValues: {
+      newOwner: '',
+    },
+  })
+
+  const web3Service = useWeb3Service()
+  const walletService = useWalletService()
+  const newOwner = watch('newOwner', '')
+
+  const { lock } = useLockData({
+    lockAddress,
+    network,
+  })
+
+  const { data: { isTransferAllowed } = {} } = useGetTransferFeeBasisPoints({
+    lockAddress,
+    network,
+  })
+
+  const maxKeysPerAddress = lock?.maxKeysPerAddress ?? 1
+
+  const onTransferFrom = async () => {
+    await walletService.transferFrom({
+      keyOwner: owner,
+      to: newOwner!,
+      lockAddress,
+      tokenId,
+    })
+  }
+
+  const transferFromMutation = useMutation(onTransferFrom)
+
+  const onTransferKeyOwnership = async () => {
+    const transferPromise = transferFromMutation.mutateAsync()
+
+    await ToastHelper.promise(transferPromise, {
+      loading: `Transferring key ownership to ${addressMinify(newOwner)}.`,
+      success: 'Key successfully transferred.',
+      error: 'There is an issue with transfer, please try again.',
+    })
+    setIsOpen(false)
+  }
+
+  useEffect(() => {
+    reset() // clear form
+  }, [isOpen, reset])
+
+  const isOwnerUnchanged = async (address: string) => {
+    return owner?.toLowerCase() === address?.toLowerCase()
+      ? 'This address is already the current owner for this key.'
+      : true
+  }
+
+  const hasMaxNumberOfKeys = async () => {
+    if (ethers.utils.isAddress(newOwner) && !!lockAddress) {
+      const total = await web3Service.totalKeys(lockAddress, newOwner, network)
+
+      return maxKeysPerAddress <= total
+        ? 'This address already owns max number of keys.'
+        : true
+    }
+    return true
+  }
+
+  const isTransferable = async () => {
```

Comment:
Rather than handle this here, you should **not** show the button on the keychain when transfers are not possible.

Reply:
this is already handled in the frontend, this is an extra check